plat/allwinner/common/sunxi_common.c

```diff
@@ -70,3 +70,32 @@ uint16_t sunxi_read_soc_id(void)
 
 	return reg >> 16;
 }
+
+/*
+ * Configure a given pin to the GPIO-OUT function and sets its level.
+ * The port is given as a capital letter, the pin is the number within
+ * this port group.
+ * To set pin PC7 to high, use: sunxi_set_gpio_out('C', 7, true);
+ */
+void sunxi_set_gpio_out(char port, int pin, bool level_high)
+{
+	uintptr_t pio_base = SUNXI_PIO_BASE;
+
+	if (port < 'A' || port > 'L')
+		return;
+	if (port == 'L')
+		pio_base = SUNXI_R_PIO_BASE;
+	else
+		pio_base += (port - 'A') * 0x24;
```

Comment:
The use of `=` vs `+=` is confusing here. How about `pio_base = SUNXI_PIO_BASE + (port - 'A') * 0x24;`, and drop the initialization earlier? And maybe `port_base`, since it's only sometimes the PIO base.